File: TestSollanFrontend/src/app/general/utils/backendMappings.ts
 export class BackendMappings {
    static readonly INDEX = '/api';
   
    static Student = class {
        static readonly ALL = "/student/all";
        static readonly SAVE_OR_UPDATE = "/student/saveOrUpdate";
        static readonly BY_ID = "/student/byId";
        static readonly BY_CLASS_ID = "/student/byClassId";
        static readonly BY_PARENT_ID = "/student/byParentId";
        static readonly BY_IDS = "/student/byIds";
        static readonly REMOVE = "/student/remove";
        static readonly BY_CLASS_IDS = "/student/byClassIds";
    }

    static Parent = class {
        static readonly ALL = "/parent/all";
        static readonly SAVE_OR_UPDATE = "/parent/saveOrUpdate";
        static readonly BY_ID = "/parent/byId";
        static readonly BY_IDS = "/parent/byIds";
        static readonly REMOVE = "/parent/remove";
        static readonly ADD_CHILDREN = "/parent/addChildren";
    }

    static Teacher = class {
        static readonly ALL = "/teacher/all";
        static readonly SAVE_OR_UPDATE = "/teacher/saveOrUpdate";
        static readonly BY_ID = "/teacher/byId";
        static readonly BY_IDS = "/teacher/byIds";
        static readonly REMOVE = "/teacher/remove";
        static readonly ADD_SUBJECT = "/teacher/addSubject";
		static readonly REMOVE_SUBJECT = "/teacher/removeSubject";
		static readonly ADD_SUBJECTS = "/teacher/addSubjects";
        static readonly REMOVE_SUBJECTS = "/teacher/removeSubjects";
        static readonly ADD_CLASS = "/teacher/addClass";
        static readonly REMOVE_CLASS = "/teacher/removeClass";
        static readonly BY_SUBJECT = "/teacher/bySubject";
        static readonly BY_CLASS_AND_SUBJECT = "/teacher/byClassAndSubject";
    }

    static Login = class {
        static readonly LOGIN = "/login"
        static readonly LOGOUT = "/logout"
        static readonly IS_AUTHENTICATED = "/isAuthenticated";
    }

    static RouterPaths = class {
        static readonly RESULT = '/result';
        static readonly QUESTION = '/question';
        static readonly HOME = '';
    }

    static StudentClass = class {
        static readonly BY_IDS = "/classes/byIds";
        static readonly ALL = "/classes/all";
        static readonly SAVE_OR_UPDATE = "/classes/saveOrUpdate";
        static readonly REMOVE = "/classes/remove";
        static readonly ADD_STUDENT = "/classes/addStudent";
        static readonly ADD_STUDENTS = "/classes/addStudents";
        static readonly ADD_SUBJECTS = "/classes/addSubjects";
        static readonly REMOVE_STUDENT = "/classes/removeStudent";
        static readonly BY_TEACHER_ID = "/classes/byTeacherId";
        static readonly BY_STUDENT_ID = "/classes/byStudentId";
        static readonly REMOVE_SUBJECT ="/classes/removeSubject";
        static readonly BY_TEACHER_AND_SUBJECT = "/classes/byTeacherAndSubject";

        
    }


    static Note = class {
        static readonly BY_STUDENT_ID = "/note/studentId";
        static readonly BY_TEACHER_ID = "/note/teacherId";
        static readonly ADD ="/note/add";
        static readonly REMOVE ="/note/remove";
        static readonly STUDENT_BY_NOTE_ID = "/note/studentByNoteId";
		static readonly TEACHER_BY_NOTE_ID = "/note/teacherByNoteId";
    }
    
    static Grade = class {
		static readonly BY_STUDENT_ID ="/grade/studentId";
		static readonly BY_STUDENT_AND_SUBJECT = "/grade/studentSubjectId";
		static readonly ADD ="/grade/add";
		static readonly REMOVE ="/grade/add";
    }
    
    static  Subject = class {
		static readonly BY_TEACHER_ID ="/subject/byTeacherId";
		static readonly BY_CLASS_ID ="/subject/byClassId";
        static readonly ADD = "/subject/add";
        static readonly ALL = "/subject/all";
		static readonly REMOVE =  "/subject/remove";
    }
    
    static ChangePassword = class {
        static readonly INDEX = "/changePassword"
        static readonly RESET = "/resetPassword"
    }

}

File: TestSollanBackend/src/main/java/com/sollan/util/BackendMappings.java
package com.sollan.util;

public class BackendMappings {
	
	
	private static final String API = "/api";
	
	
	public class StudentClass{
		
		public static final String ALL = API+"/classes/all";
		public static final String BY_IDS = API + "/classes/byIds";
		public static final String SAVE_OR_UPDATE = API +"/classes/saveOrUpdate";
		public static final String REMOVE = API +"/classes/remove";
		public static final String ADD_STUDENT = API +"/classes/addStudent";
		public static final String ADD_STUDENTS = API +"/classes/addStudents";
		public static final String ADD_SUBJECTS = API +"/classes/addSubjects";
		public static final String REMOVE_STUDENT = API +"/classes/removeStudent";
		public static final String ADD_SUBJECT = API +"/classes/addSubject";
		public static final String REMOVE_SUBJECT = API +"/classes/removeSubject";
		public static final String BY_TEACHER_ID = API +"/classes/byTeacherId";
		public static final String BY_TEACHER_AND_SUBJECT = API +"/classes/byTeacherAndSubject";
		public static final String BY_STUDENT_ID = API +"/classes/byStudentId";
	}
	
	public class Student{
		public static final String ALL = API +"/student/all";
		public static final String BY_ID = API +"/student/byId";
		public static final String BY_CLASS_ID = API +"/student/byClassId";
		public static final String BY_PARENT_ID = API +"/student/byParentId";
		public static final String BY_IDS = API +"/student/byIds";
		public static final String SAVE_OR_UPDATE = API +"/student/saveOrUpdate";
		public static final String REMOVE = API +"/student/remove";
		public static final String BY_CLASS_IDS = API +"/student/byClassIds";
	}
	
	public class Teacher{
		public static final String ALL = API + "/teacher/all";
		public static final String BY_ID = API +"/teacher/byId";
		public static final String SAVE_OR_UPDATE = API +"/teacher/saveOrUpdate";
		public static final String REMOVE = API +"/teacher/remove";
		public static final String ADD_SUBJECT = API +"/teacher/addSubject";
		public static final String REMOVE_SUBJECT = API +"/teacher/removeSubject";
		public static final String ADD_SUBJECTS = API +"/teacher/addSubjects";
		public static final String REMOVE_SUBJECTS = API +"/teacher/removeSubjects";
		public static final String ADD_CLASS = API +"/teacher/addClass";
		public static final String REMOVE_CLASS = API +"/teacher/removeClass";
		public static final String BY_SUBJECT = API +"/teacher/bySubject";
		public static final String BY_CLASS_AND_SUBJECT = API +"/teacher/byClassAndSubject";
	}
	
	public class Parent {
		public static final String ALL = API + "/parent/all";
		public static final String BY_ID = API +"/parent/byId";
		public static final String SAVE_OR_UPDATE = API +"/parent/saveOrUpdate";
		public static final String REMOVE = API +"/parent/remove";
		public static final String ADD_CHILDREN = API +"/parent/addChildren";
	}

	
	public class Login {
		public static final String LOGIN = API + "/login";
		public static final String LOGOUT = API + "/logout";
		public static final String IS_AUTHENTICATED = API + "/isAuthenticated";
	}
	
	public class Note{
		public static final String BY_STUDENT_ID = API + "/note/studentId";
		public static final String BY_TEACHER_ID = API + "/note/teacherId";
		public static final String ADD = API + "/note/add";
		public static final String REMOVE = API + "/note/remove";
		public static final String STUDENT_BY_NOTE_ID = API + "/note/studentByNoteId";
		public static final String TEACHER_BY_NOTE_ID = API + "/note/teacherByNoteId";
	}
	
	public class Grade{
		public static final String BY_STUDENT_ID = API + "/grade/studentId";
		public static final String BY_STUDENT_AND_SUBJECT = API + "/grade/studentSubjectId";
		public static final String ADD = API + "/grade/add";
		public static final String REMOVE = API + "/grade/add";
		public static final String DOWNLOAD_GRADES = API + "/grade/download";
	}
	
	public class Subject{
		public static final String BY_TEACHER_ID = API + "/subject/byTeacherId";
		public static final String BY_CLASS_ID = API + "/subject/byClassId";
		public static final String ADD = API + "/subject/add";
		public static final String REMOVE = API + "/subject/remove";
		public static final String ALL = API + "/subject/all";
	}
	
	 public class  ChangePassword  {
		 public static final String INDEX = API + "/changePassword";
		 public static final String RESET = API + "/resetPassword";
	}
	 
	 public class  Message  {
		 public static final String SEND_MESSAGE = API + "/message/sendMessage";
		 public static final String SEND_STATEMENT = API + "/message/sendStatement";
		 public static final String GET_STATEMENTS = API + "/message/getStatements";
		 public static final String DELETE_MESSAGE = API + "/message/deleteMessage";
		 public static final String BY_TEACHER_ID = API + "/message/byTeacherId";
		 public static final String BY_STUDENT_ID = API + "/message/byStudentId";
		 public static final String BY_PARENT_ID = API + "/message/byParentId";
		 public static final String BY_TYPE_AND_ID = API + "/message/byTypeAndId";
		 public static final String SET_OPENED = API + "/message/setOpened";
	}
	 
	 public class Attendance {
		 public static final String BY_STUDENT_ID = API + "/attendance/byStudentId";
		 public static final String SAVE = API + "/attendance/save";
		 public static final String UPDATE_ATTENDANCE = API + "/attendance/update";
		 public static final String BY_STUDENT_ID_AND_DATE = API + "/attendance/byIdAndDate";
		 public static final String BY_STUDENT_SUBJECT_AND_DATE = API + "/attendance/byStudentSubjectDate";
		 public static final String FROM_THIS_YEAR = API + "/attendance/fromThisYear";
		 
	 }
	 
	 public class Post {
		 public static final String ALL = API + "/post/all";
		 public static final String SAVE = API + "/post/save";
		 public static final String REMOVE = API + "/post/remove";
	 }
	 

}
